lolly/data/lolly_string.ipp

```diff
@@ -33,17 +34,54 @@ lolly_string_rep<T>::resize (int m) {
   if (mm != nn) {
     if (mm != 0) {
       if (nn != 0) {
-        a= tm_resize_array<T> (mm, a);
+        a_N= mm;
+        a     = tm_resize_array<T> (mm, a);
       }
       else {
-        a= tm_new_array<T> (mm);
+        a_N= mm;
+        a     = tm_new_array<T> (mm);
       }
     }
-    else if (nn != 0) tm_delete_array (a);
+    else if (nn != 0) {
+      a_N= 0;
+      tm_delete_array (a);
```

Comment:
why we do not need to set `a= NULL` here?

Reply:
in this branch, we resize the string to 0

Reply:
old code does not set `a= AULL`